aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -724,13 +724,41 @@ Tensor& normal_out_cuda(Tensor& output, double mean, const Tensor& std, Generato
 }
 
 Tensor& normal_out_cuda(Tensor& output, const Tensor& mean, const Tensor& std, Generator* gen) {
+  bool expandable = are_expandable(mean.sizes(), std.sizes());
+  bool empty_output = output.numel() == 0;
+
+  if (expandable) {
+    auto shape = at::infer_size(mean.sizes(), std.sizes());
+    TORCH_CHECK(empty_output || output.sizes().equals(shape), "output size is not the same as broadcasted size of mean and std");
```

Comment:
nit (not blocking): It's nice to report the sizes (output size, size of mean, size of std) so users can correlate the function call back to the inputs.